platforms/cuda/src/CudaKernels.cpp

```diff
@@ -2045,8 +2185,80 @@ double CudaCalcNonbondedForceKernel::execute(ContextImpl& context, bool includeF
             cuEventRecord(pmeSyncEvent, pmeStream);
             cu.restoreDefaultStream();
         }
+
+        // As written, we check only the Electrostatic grid pointer to get here.  We could separate them out, but for
+        // now we assume that LJPME can only be used if electrostatic PME is also active.
+        if(doLJPME){
+            if (usePmeStream)
+                cu.setCurrentStream(dispersionPmeStream);
```

Comment:
Sounds good.  I agree about the nasty code duplication, and would rather avoid it.  If we pack C6 into a posq-like array, the spreading and probing codes become identical, but for the grid size macros.